rest_framework/utils/field_mapping.py

```diff
@@ -102,7 +102,7 @@ def get_field_kwargs(field_name, model_field):
     if model_field.null and not isinstance(model_field, models.NullBooleanField):
         kwargs['allow_null'] = True
 
-    if model_field.blank:
+    if model_field.blank and isinstance(model_field, models.CharField):
```

Comment:
I'd suggest having a look through the history for this line first.
Have we ever had different behavior in the past?
Are there any other built-in fields that don't subclass CharField but that do take an `allow_blank` argument?